From 678b7168ad9ef36907088df146c2990ea4258bfd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Wed, 16 Jul 2014 13:00:42 +0800
Subject: [PATCH] ASoC: rockchip: rk30_i2s.c use pinctrl_pm_select_sleep_state

call pinctrl_select_state could trigger Oops when suspend
if pinctrl no configured.
---
 sound/soc/rockchip/rk30_i2s.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/rockchip/rk30_i2s.c b/sound/soc/rockchip/rk30_i2s.c
index 72784ab2058a..cb5d532f1fcd 100755
--- a/sound/soc/rockchip/rk30_i2s.c
+++ b/sound/soc/rockchip/rk30_i2s.c
@@ -494,14 +494,14 @@ static int rockchip_i2s_suspend_noirq(struct device *dev)
 {
 	I2S_DBG("Enter %s, %d\n", __func__, __LINE__);
 
-	return pinctrl_select_state(dev->pins->p, dev->pins->sleep_state);
+	return pinctrl_pm_select_sleep_state(dev);
 }
 
 static int rockchip_i2s_resume_noirq(struct device *dev)
 {
 	I2S_DBG("Enter %s, %d\n", __func__, __LINE__);
 
-	return pinctrl_select_state(dev->pins->p, dev->pins->default_state);
+	return pinctrl_pm_select_default_state(dev);
 }
 #else
 #define rockchip_i2s_suspend_noirq NULL
-- 
2.35.3

